### ISSUE TITLE
[Bug]: Controlnet appears in Auto-1111 interface but does not seem to work

### ISSUE THREAD

**Issue by mrmeseeks23** [Mikubill/sd-webui-controlnet#2017]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

Able to generate images as normal but no control net output.  Associated error noted below.
OS: Linux/Ubuntu 22.04
Pytorch 2.0.1 Cuda 11.8, Python 10.6

### Steps to reproduce the problem

1. Open Automatic 1111
2. Enable controlnet 1.1
3. Generate txt-img

### What should have happened?

images output c/w controlnet settings

### Commit where the problem happens

webui: Automatic-1111, latest update
controlnet: v1.1

### What browsers do you use to access the UI ?

Google Chrome, Apple Safari, Microsoft Edge

### Command Line Arguments

```Shell
launch the gui (no issue) with ./webui.sh on Linux/Ubuntu 22.04
```

### List of enabled extensions

Deforum, open pose editor, clip-interrogator-ext, LORA

### Console logs

```Shell
Total progress: 100%|████████████████████████████████████████████████| 20/20 [00:01<00:00, 19.96it/s]
2023-09-01 04:12:24,606 - ControlNet - INFO - Loading model: control_v11f1p_sd15_depth [e3b0c442]t/s]
*** Error running process: /workspace/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py
    Traceback (most recent call last):
      File "/workspace/stable-diffusion-webui/modules/scripts.py", line 619, in process
        script.process(p, *script_args)
      File "/workspace/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 738, in process
        model_net = Script.load_control_model(p, unet, unit.model, unit.low_vram)
      File "/workspace/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 311, in load_control_model
        model_net = Script.build_control_model(p, unet, model, lowvram)
      File "/workspace/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 339, in build_control_model
        state_dict = load_state_dict(model_path)
      File "/workspace/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/utils.py", line 23, in load_state_dict
        torch.load(ckpt_path, map_location=torch.device(location))
      File "/workspace/stable-diffusion-webui/modules/safe.py", line 108, in load
        return load_with_extra(filename, *args, extra_handler=global_extra_handler, **kwargs)
      File "/workspace/stable-diffusion-webui/modules/safe.py", line 156, in load_with_extra
        return unsafe_torch_load(filename, *args, **kwargs)
      File "/workspace/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/serialization.py", line 815, in load
        return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
      File "/workspace/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/serialization.py", line 1033, in _legacy_load
        magic_number = pickle_module.load(f, **pickle_load_args)
    _pickle.UnpicklingError: invalid load key, 'v'.
```

### Additional information

Images generating (as they normally would using basic txt2img) but no CN involvement.  Associated with above error.  I have downloaded the associated .pth and .yaml files.  I have tried with models located in both the extension directory and within the 'stable-diffusion-webui/models/controlnet directory'.

**Comment by aipicsstudio** [comment]
I got the same issue since yesterday's update. The problem remain even after restarting the computer. I test with all other extensions disabled. I even trying installing python and automatic1111 from scratch

**Comment by aipicsstudio** [comment]
In my case, the problem occures on deforums but does not occur on img2img

**Comment by aipicsstudio** [comment]
![deforum error screenshot 2](https://github.com/Mikubill/sd-webui-controlnet/assets/143832355/b53ba633-59d4-49e4-a6a8-8cabe419540b)
![deforum error screenshot 1](https://github.com/Mikubill/sd-webui-controlnet/assets/143832355/a63e4872-5f45-4a38-ac56-3b0d9324179d)

**Comment by IlonaKova** [comment]
I have the same problem after the update. The controlnet does not work in the Deforum

**Comment by TheRealCrew** [comment]
Same problem since yesterday after updating to SD V1.6.0.

*** Error running setup: E:\Stable_diffusion\stable-diffusion-webui\modules/processing_scripts\refiner.py
    Traceback (most recent call last):
      File "E:\Stable_diffusion\stable-diffusion-webui\modules\scripts.py", line 740, in setup_scrips
        script_args = p.script_args[script.args_from:script.args_to]
    TypeError: unhashable type: 'slice'

---
*** Error running setup: E:\Stable_diffusion\stable-diffusion-webui\modules/processing_scripts\seed.py
    Traceback (most recent call last):
      File "E:\Stable_diffusion\stable-diffusion-webui\modules\scripts.py", line 740, in setup_scrips
        script_args = p.script_args[script.args_from:script.args_to]
    TypeError: unhashable type: 'slice'

---
*** Error running setup: E:\Stable_diffusion\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py
    Traceback (most recent call last):
      File "E:\Stable_diffusion\stable-diffusion-webui\modules\scripts.py", line 740, in setup_scrips
        script_args = p.script_args[script.args_from:script.args_to]
    TypeError: unhashable type: 'slice'

---
*** Error running setup: E:\Stable_diffusion\stable-diffusion-webui\extensions-builtin\extra-options-section\scripts\extra_options_section.py
    Traceback (most recent call last):
      File "E:\Stable_diffusion\stable-diffusion-webui\modules\scripts.py", line 740, in setup_scrips
        script_args = p.script_args[script.args_from:script.args_to]
    TypeError: unhashable type: 'slice'

---
2023-09-02 11:57:22,653 - ControlNet - WARNING - No ControlNetUnit detected in args. It is very likely that you are having an extension conflict.Here are args received by ControlNet: [].
2023-09-02 11:57:22,653 - ControlNet - WARNING - No ControlNetUnit detected in args. It is very likely that you are having an extension conflict.Here are args received by ControlNet: [].

**Comment by TheRealCrew** [comment]
Rolled back to the sd-webui version 1.5.2. Deforum is working with the older version. Add "git checkout v1.5.2" to your webui-user.bat

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/37497418/d32aa82a-0e88-4573-812f-3b2ef4e1bfe9)

**Comment by IlonaKova** [comment]
The method described above does not work for Google Collab users.

**Comment by Nodam1** [comment]
It also doesnt work for me in the Automatic webui..

**Comment by esirt** [comment]
i had similar issue . and after deleting / installing several time using the extension tab the issue remained . some time it was working and sometime didn't . finally it got fixed by deleting the control net folder and then cloning the repo using command line into extension folder . and then ran the requirement.txt using activated enve .

it was giving me that warning error . the extension wasn't working properly also another weird side effect was on extension tab i had to press update to just see my extensions . / in lora ,checkpoint tabs i had to press refresh to see the items . and even after seeing the models the loading display wouldn't go away so gray mask all over the screen .

**Comment by luckybit4755** [comment]
In my case the error looks like this:

```
File "stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet_ui/controlnet_ui_group.py", line 600, in filter_selected
) = global_state.select_control_type(k)
File "stable-diffusion-webui/extensions/sd-webui controlnet/scripts/global_state.py", line 242, in select_control_type
     default_option = preprocessor_filters[control_type]
TypeError: unhashable type: 'list'
```

The labels on the control types are also weirdly duplicated:

![controlnet-dupes](https://github.com/Mikubill/sd-webui-controlnet/assets/20212234/a09484a7-82b5-4625-8e1a-d6b959d4ac1c)

It appears to be passing values like ["Lineart","Lineart"] instead of just "Lineart" to select_control_type. I tried hacking it to check for instance of List and just take the first argument but the ui for loading the Preprocessor and Model just spins and spins.

The ui says it is "ControlNet v1.1.409"

**Comment by mrmeseeks23** [comment]
Yeah mine doesn't give any error in the GUI for Auto1111.  It will render teh images in the T2I window, it will even generate the mask/'preview' of the control net with an example image in place.  Also of note I can no longer drag and drop from the output pane to the control net area.

**Comment by inewlife** [comment]
INFO:httpx:HTTP Request: POST http://localhost:8000/api/predict "HTTP/1.1 200 OK"
*** Error running process: /code/stable-diffusion-webui/data-oss/extensions/sd-webui-controlnet/scripts/controlnet.py
    Traceback (most recent call last):
      File "/code/stable-diffusion-webui/modules/scripts.py", line 495, in process
        script.process(p, *script_args)
      File "/code/stable-diffusion-webui/data-oss/extensions/sd-webui-controlnet/scripts/controlnet.py", line 977, in process
        self.controlnet_hack(p)
      File "/code/stable-diffusion-webui/data-oss/extensions/sd-webui-controlnet/scripts/controlnet.py", line 966, in controlnet_hack
        self.controlnet_main_entry(p)
      File "/code/stable-diffusion-webui/data-oss/extensions/sd-webui-controlnet/scripts/controlnet.py", line 688, in controlnet_main_entry
        model_net = Script.load_control_model(p, unet, unit.model)
      File "/code/stable-diffusion-webui/data-oss/extensions/sd-webui-controlnet/scripts/controlnet.py", line 321, in load_control_model
        model_net = Script.build_control_model(p, unet, model)
      File "/code/stable-diffusion-webui/data-oss/extensions/sd-webui-controlnet/scripts/controlnet.py", line 331, in build_control_model
        raise RuntimeError(f"You have not selected any ControlNet Model.")
    RuntimeError: You have not selected any ControlNet Model.

**Comment by cheezecrisp** [comment]
> In my case the error looks like this:
> 
> ```
> File "stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet_ui/controlnet_ui_group.py", line 600, in filter_selected
> ) = global_state.select_control_type(k)
> File "stable-diffusion-webui/extensions/sd-webui controlnet/scripts/global_state.py", line 242, in select_control_type
>      default_option = preprocessor_filters[control_type]
> TypeError: unhashable type: 'list'
> ```
> 
> The labels on the control types are also weirdly duplicated:
> 
> ![controlnet-dupes](https://user-images.githubusercontent.com/20212234/266787712-a09484a7-82b5-4625-8e1a-d6b959d4ac1c.png)
> 
> It appears to be passing values like ["Lineart","Lineart"] instead of just "Lineart" to select_control_type. I tried hacking it to check for instance of List and just take the first argument but the ui for loading the Preprocessor and Model just spins and spins.
> 
> The ui says it is "ControlNet v1.1.409"

Same problem. Not only controlnet but alse some other extensions e.g. animatediff

**Comment by UncleK** [comment]
> > In my case the error looks like this:
> > ```
> > File "stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet_ui/controlnet_ui_group.py", line 600, in filter_selected
> > ) = global_state.select_control_type(k)
> > File "stable-diffusion-webui/extensions/sd-webui controlnet/scripts/global_state.py", line 242, in select_control_type
> >      default_option = preprocessor_filters[control_type]
> > TypeError: unhashable type: 'list'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The labels on the control types are also weirdly duplicated:
> > ![controlnet-dupes](https://user-images.githubusercontent.com/20212234/266787712-a09484a7-82b5-4625-8e1a-d6b959d4ac1c.png)
> > It appears to be passing values like ["Lineart","Lineart"] instead of just "Lineart" to select_control_type. I tried hacking it to check for instance of List and just take the first argument but the ui for loading the Preprocessor and Model just spins and spins.
> > The ui says it is "ControlNet v1.1.409"
> 
> Same problem. Not only controlnet but alse some other extensions e.g. animatediff

I encountered this too, deleting venv folder didn't help,
only happened in "selecting "Control Type" ", manually selections of Preprocessor and models are okay.

**Comment by huchenlei** [comment]
See https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/12886
The issue has been fixed in lastest version of WebUI.

**Comment by huchenlei** [comment]
Fix PR:
https://github.com/richrobber2/canvas-zoom/issues/92

**Comment by sinoku2024** [comment]
Why was this bug closed? The fix for bug [#12886](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/12886) mentionned by @huchenlei has nothing to do with the ops bug report!

I experience exactly the same problem as the op (see CONSOLE ERROR 3 in point 8 below) but on macos Sonoma (14.2.1) on Apple silicon (Apple M2 Ultra) and it can be reproduced easily as follows:

1. Install Stable Diffusion on Macos

```
brew install cmake protobuf pyenv rust git wget
cd $HOME
git clone https://github.com/AUTOMATIC1111/stable-diffusion-webui
cd stable-diffusion-webui
pyenv install 3.10
pyenv local 3.10.13
eval "$(pyenv init -)"
./webui.sh
```

2. This leads to **CONSOLE ERROR 1** (likely unrelated):

```
Warning: caught exception 'Torch not compiled with CUDA enabled', memory monitor disabled
==============================================================================
You are running torch 2.1.0.
The program is tested to work with torch 2.1.2.
To reinstall the desired version, run with commandline flag --reinstall-torch.
Beware that this will cause a lot of large files to be downloaded, as well as
there are reports of issues with training tab on the latest version.

Use --skip-version-check commandline argument to disable this check.
==============================================================================
```

3.  In Web UI: Extensions > Install from URL tab > https://github.com/Mikubill/sd-webui-controlnet > Install

4. Install ControlNet Models:

```
cd ~/stable-diffusion-webui/extensions/sd-webui-controlnet/models
git clone https://huggingface.co/lllyasviel/ControlNet-v1-1
```

5. @WebUI: Extensions > Installled > Apply and Restart UI


6. This leads to **CONSOLE ERROR 2** (likely unrelated):

```
ControlNet init warning: Unable to install insightface automatically. Please try run `pip install insightface` manually.
```


7. Now follow the steps on https://stable-diffusion-art.com/controlnet/#Using_ControlNet_8211_a_simple_example


8. After pressing the Generate button you get **CONSOLE ERROR 3 (this bug)**

```
2024-03-10 20:48:36,914 - ControlNet - INFO - Loading model: control_v11p_sd15_openpose [e3b0c442] (model_path=/Users/sinoku/stable-diffusion-webui/extensions/sd-webui-controlnet/models/ControlNet-v1-1/control_v11p_sd15_openpose.pth)
*** Error running process: /Users/sinoku/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py
    Traceback (most recent call last):
      File "/Users/sinoku/stable-diffusion-webui/modules/scripts.py", line 784, in process
        script.process(p, *script_args)
      File "/Users/sinoku/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 1279, in process
        self.controlnet_hack(p)
      File "/Users/sinoku/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 1264, in controlnet_hack
        self.controlnet_main_entry(p)
      File "/Users/sinoku/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 936, in controlnet_main_entry
        model_net, control_model_type = Script.load_control_model(p, unet, unit.model)
      File "/Users/sinoku/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 367, in load_control_model
        control_model = Script.build_control_model(p, unet, model)
      File "/Users/sinoku/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 395, in build_control_model
        state_dict = load_state_dict(model_path)
      File "/Users/sinoku/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/utils.py", line 22, in load_state_dict
        state_dict = unsafe_torch_load(ckpt_path, map_location=torch.device(location))
      File "/Users/sinoku/stable-diffusion-webui/env/lib/python3.10/site-packages/torch/serialization.py", line 1028, in load
        return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
      File "/Users/sinoku/stable-diffusion-webui/env/lib/python3.10/site-packages/torch/serialization.py", line 1246, in _legacy_load
        magic_number = pickle_module.load(f, **pickle_load_args)
    _pickle.UnpicklingError: invalid load key, 'v'.
```

As mentionned by the op in the bug report image generation works but ControlNet settings have absolutely no influence! So I'd say this bug still exists!
